### PR TITLE
QoL changes to the queue manager and whats reported

### PR DESCRIPTION
**Pull request by Lnaden**
## Description
This PR fixes a few issues relating to more verbose reports from the queue managers as to the pass/fail rate of jobs, a verbosity toggle, and you can now just scale up in dask rather than adapt.

Fixes

## Status
- [ ] Changelog updated
- [x] Ready to go

### CODE REVIEW

File: qcfractal/queue/managers.py [hunk 1]
```diff
@@ -321,20 +321,26 @@ def update(self, new_tasks: bool=True) -> bool:
...
+        n_fail = 0
+        n_result = len(results)
+        if n_result:
             payload = self._payload_template()
 
             # Upload new results
             payload["data"] = results
             try:
-                response = self.client._automodel_request("queue_manager", "post", payload)
+                _ = self.client._automodel_request("queue_manager", "post", payload)
```
Comment (dgasmith) [inline qcfractal/queue/managers.py]: Just remove `_ = ` here if not used.

File: qcfractal/queue/managers.py [hunk 2]
```diff
@@ -321,20 +321,26 @@ def update(self, new_tasks: bool=True) -> bool:
...
             except IOError:
                 # TODO something as we didnt successfully add the data
                 self.logger.warning("Post complete tasks was not successful. Data may be lost.")
 
-            self.active -= len(results)
+            self.active -= n_result
+            n_success = [r.success for r in results.values()].count(True)
```
Comment (dgasmith) [inline qcfractal/queue/managers.py]: `sum([r.success for r in results.values()])`, but thats the NumPy in me.

**Comment by dgasmith** [comment]
We also need print the first error message in `qcfractal-manager --test`.

File: qcfractal/queue/managers.py (hunk 2 above)
Comment (Lnaden) [reply qcfractal/queue/managers.py]: That is what I was originally going to do as well, but from what I was reading this is actually just a bit faster:

https://stackoverflow.com/a/12766783/10364409

Comment (dgasmith) [reply qcfractal/queue/managers.py]: For items like this speed isn't an issue so I would go with which is more clear. No real preference.

**Comment by Lnaden** [comment]
> We also need print the first error message in qcfractal-manager --test.

So I chose to print the last error message instead because it was a one liner with nothing to track. I can do first if thats what you would prefer.

https://github.com/MolSSI/QCFractal/pull/234/files#diff-12fe35ea3c88dd77f827cc649ee2df2dR476

File: qcfractal/queue/managers.py [hunk 3]
```diff
@@ -468,6 +473,7 @@ def test(self, n=1) -> bool:
 
         if failures:
             self.logger.error("{}/{} tasks failed!".format(failures, len(results)))
+            self.logger.error(f"Last known error to help you debug it:\n{result.error.error_message}")
```
Comment (dgasmith) [inline qcfractal/queue/managers.py]: Makes me a little uncomfortable view a iterator depends, it also doesn't work for multiple programs in the test.

**Comment by Lnaden** [comment]
@dgasmith 

Trying to adjust the server looks like it broke a few other things. I'm going to hold off on further tinkering until you get back to this since I don't know what you did server side and don't want to fix the wrong thing, but here is the behavior (I set both in the tests here for you to see):

If the `specs` key in TorsionDriveDataset are `b3lyp-d3`, the following error is thrown:
```python
    @using_plotly
    def test_plot_torsiondrive_dataset(TDDSFixture):
        client, ds = TDDSFixture
    
>       ds.visualize("c1ccc(cc1)N-[3, 5, 6, 12]", ["b3lyp-d3", "uff"], units="kJ / mol", return_figure=True)

tests/test_visualization.py:88: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
collections/torsiondrive_dataset.py:408: in visualize
    td = self.df.loc[index, spec]
../../../../miniconda3/envs/qcf/lib/python3.7/site-packages/pandas/core/indexing.py:1494: in __getitem__
    return self._getitem_tuple(key)
../../../../miniconda3/envs/qcf/lib/python3.7/site-packages/pandas/core/indexing.py:868: in _getitem_tuple
    return self._getitem_lowerdim(tup)
../../../../miniconda3/envs/qcf/lib/python3.7/site-packages/pandas/core/indexing.py:1017: in _getitem_lowerdim
    return getattr(section, self.name)[new_key]
../../../../miniconda3/envs/qcf/lib/python3.7/site-packages/pandas/core/indexing.py:1500: in __getitem__
    return self._getitem_axis(maybe_callable, axis=axis)
../../../../miniconda3/envs/qcf/lib/python3.7/site-packages/pandas/core/indexing.py:1913: in _getitem_axis
    return self._get_label(key, axis=axis)
../../../../miniconda3/envs/qcf/lib/python3.7/site-packages/pandas/core/indexing.py:137: in _get_label
    return self.obj._xs(label, axis=axis)
../../../../miniconda3/envs/qcf/lib/python3.7/site-packages/pandas/core/generic.py:3585: in xs
    loc = self.index.get_loc(key)
../../../../miniconda3/envs/qcf/lib/python3.7/site-packages/pandas/core/indexes/base.py:2659: in get_loc
    return self._engine.get_loc(self._maybe_cast_indexer(key))
pandas/_libs/index.pyx:108: in pandas._libs.index.IndexEngine.get_loc
    ???
pandas/_libs/index.pyx:132: in pandas._libs.index.IndexEngine.get_loc
    ???
pandas/_libs/hashtable_class_helper.pxi:1601: in pandas._libs.hashtable.PyObjectHashTable.get_item
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   KeyError: 'b3lyp-d3'

pandas/_libs/hashtable_class_helper.pxi:1608: KeyError
```

However, if the `spec` has "B3LYP-D3" its a different error:
```python
___________________________________________________________________________ test_plot_torsiondrive_dataset_measured ____________________________________________________________________________

TDDSFixture = (<qcfractal.interface.client.FractalClient object at 0x124b91eb8>, <TorsionDriveDataset(name=`OpenFF Fragmenter Phenyl Benchmark`, id='5c9526afb6a2de5f195ee874', client='https://api.qcarchive.molssi.org:443/') >)

    @using_plotly
    def test_plot_torsiondrive_dataset_measured(TDDSFixture):
        client, ds = TDDSFixture
    
>       ds.visualize("c1ccc(cc1)N-[3, 5, 6, 12]", "B3LYP-D3", units="kJ / mol", use_measured_angle=True, return_figure=True)

tests/test_visualization.py:97: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <TorsionDriveDataset(name=`OpenFF Fragmenter Phenyl Benchmark`, id='5c9526afb6a2de5f195ee874', client='https://api.qcarchive.molssi.org:443/') >, entries = ['c1ccc(cc1)N-[3, 5, 6, 12]']
specs = ['B3LYP-D3'], relative = True, units = 'kJ / mol', digits = 3, use_measured_angle = True, return_figure = True

    def visualize(self,
                  entries: Union[str, List[str]],
                  specs: Union[str, List[str]],
                  relative: bool=True,
                  units: str="kcal / mol",
                  digits: int=3,
                  use_measured_angle: bool=False,
                  return_figure: Optional[bool]=None) -> 'plotly.Figure':
        """
        Parameters
        ----------
        entries : Union[str, List[str]]
            A single or list of indices to plot.
        specs : Union[str, List[str]]
            A single or list of specifications to plot.
        relative : bool, optional
            Shows relative energy, lowest energy per scan is zero.
        units : str, optional
            The units of the plot.
        digits : int, optional
            Rounds the energies to n decimal places for display.
        use_measured_angle : bool, optional
            If True, the measured final angle instead of the constrained optimization angle.
            Can provide more accurate results if the optimization was ill-behaved,
            but pulls additional data from the server and may take longer.
        return_figure : Optional[bool], optional
            If True, return the raw plotly figure. If False, returns a hosted iPlot. If None, return a iPlot display in Jupyter notebook and a raw plotly figure in all other circumstances.
    
        Returns
        -------
        plotly.Figure
            Description
    
        Raises
        ------
        TypeError
            Description
        """
    
        show_spec = True
        if isinstance(specs, str):
            specs = [specs]
            show_spec = False
    
        if isinstance(entries, str):
            entries = [entries]
    
        # Query all of the specs and make sure they are valid
        for spec in specs:
            self.query(spec)
    
        traces = []
    
        xmin = 1e12
        xmax = -1e12
        # Loop over specifications
        for spec in specs:
            # Loop over indices (groups colors by entry)
            for index in entries:
    
                record = self.get_entry(index)
    
                td = self.df.loc[index, spec]
                min_energy = 1e12
    
                # Pull the dict apart
                x = []
                y = []
>               for k, v in td.final_energies().items():
E               AttributeError: 'float' object has no attribute 'final_energies'

collections/torsiondrive_dataset.py:414: AttributeError
```

**Comment by dgasmith** [comment]
Make `UFF` and `B3LYP-D3` in caps. I have this fixed in my feature branch.

**Comment by Lnaden** [comment]
That fixes the first problem locally, but not the second where the td seems to get set to a float?

**Comment by dgasmith** [comment]
I am not sure. I will review and merge and patch in another PR.

**Pull request merged**